### PR TITLE
drone Allow using a non-default kubernetes service account

### PR DESCRIPTION
**Pull request by james-mcgoodwin**
This PR lets the user choose to not use the built in 'default' Kubernetes service account in the namespace drone is being deployed to. 

Our use case for this is we create specific KSA's for each of our services, while optionally attaching it to GCPs Workload Identity system via annotations. 

If left unchanged by the user, the default KSA will still be used and no new KSA will be generated. 

If the user overrides the serviceAccount: object they can do the following: 

* name: [string] - Select the name of a KSA to use instead of 'default'
* create: [boolean] - Optionally create a new KSA  if it does not exist
* annotations: [object] - Apply service account annotations to the optionally created KSA

Note that annotations will not be applied to any KSA if the create bool is false.

I feel like this PR dove tales with the [Allow custom webserver port](https://github.com/drone/charts/pull/21) PR. 

My ultimate goal is a drone running with out root (using port 8080), with a KSA that we choose (named predictably 'drone')

### CODE REVIEW

**Review by stekole — APPROVED** [review]

**Comment by james-mcgoodwin** [comment]
It looks like the CI linting system is out of commission, it's failing for other PR's on the bitnami charts URL. 

```
Error: looks like "https://charts.bitnami.com" is not a valid chart repository or cannot be reached: failed to fetch https://charts.bitnami.com/index.yaml : 403 Forbidden
--
```

**Comment by james-mcgoodwin** [comment]
Popping this commit message out since it changes a default value in values.yaml: 

> CI fails to pass because values.yaml contains an invalid null string.
> Running 'helm lint ./drone' fails
> Even calling helm lint with an additional
'-f ./drone/ci/test-values.yaml' still fails to validate values.yaml

    $  helm lint ./drone/ -f ./drone/ci/test-values.yaml
    ==> Linting ./drone/
    [ERROR] values.yaml: - env.DRONE_SERVER_HOST: String length must be greater than or equal to 3

    Error: 1 chart(s) linted, 1 chart(s) failed

> I have updated values.yaml with the example string used in the drone
> documentation page for the DRONE_SERVER_HOST value:

File: charts/drone/templates/deployment.yaml [hunk 1]
```diff
@@ -29,6 +29,8 @@ spec:
       automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      serviceAccount: {{ .Values.serviceAccount.name }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
```
Comment (cognifloyd) [inline charts/drone/templates/deployment.yaml:33]:
```suggestion
      serviceAccountName: {{ .Values.serviceAccount.name }}
```
I think you only need `serviceAccountName`.

**Review by cognifloyd — COMMENTED** [review]
This needs to be updated with master.

File: charts/drone/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.serviceAccount.create }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.serviceAccount.name }}
+  annotations:
+{{ toYaml .Values.serviceAccount.annotations | indent 4 }}
```
Comment (cognifloyd) [inline charts/drone/templates/serviceaccount.yaml:8]: I like using `nindent` to keep the template indentation aligned with the yaml indentation.
```suggestion
    {{- toYaml .Values.serviceAccount.annotations | nindent 4 }}
```

File: charts/drone/values.schema.json [hunk 3]
```diff
@@ -66,6 +67,24 @@
...
+		  "name": {
+			  "$id": "#/properties/serviceAccount/name",
+			  "type": "string"
+		  },
+		  "create": {
+			  "$id": "#/properties/serviceAccount/create",
+			  "type": "boolean"
+		  }
+	  } 
+    },
```
Comment (cognifloyd) [inline charts/drone/values.schema.json:87]: whitespacing is inconsistent. You've got tabs and a trailing space.
```suggestion
    "serviceAccount": {
      "$id": "#/properties/serviceAccount",
      "type": "object",
      "required": [
        "name",
        "create"
      ],
      "properties": {
        "name": {
          "$id": "#/properties/serviceAccount/name",
          "type": "string"
        },
        "create": {
          "$id": "#/properties/serviceAccount/create",
          "type": "boolean"
        }
      }
    },
```

**Comment by Ashtonian** [comment]
would like this to use a sa with server for eks irsa for s3 access.

**Comment by bkk-bcd** [comment]
> would like this to use a sa with server for eks irsa for s3 access.

We need this for the same reason. Helm create generates all the service account goodness one needs, it shouldn't have been removed to begin with.

Can we get this merged?

**Comment by bkk-bcd** [comment]
Thanks guys. If @jimsheldon is busy I can send up another PR

**Comment by jimsheldon** [comment]
#76 has been merged, closing this out

**Pull request closed without merging**